sway/desktop/render.c

```diff
@@ -180,6 +165,41 @@ static void render_surface_iterator(struct sway_output *output,
 	scale_box(&dst_box, wlr_output->scale);
 
 	data->deco_data.corner_radius *= wlr_output->scale;
+
+	// Render blur
+	struct decoration_data deco_data = data->deco_data;
+	bool opaque = true;
+	if (view && view->surface) {
+		struct wlr_gles2_texture_attribs attribs;
+		wlr_gles2_texture_get_attribs(view->surface->buffer->texture, &attribs);
+		opaque = (view->surface->opaque || !attribs.has_alpha);
+	}
+	if (view
+			// Don't render blur for subsurfaces
+			&& view->surface == surface
+			&& deco_data.blur
+			&& view->container->blur_enabled
+			// Check if window has alpha
+			&& !(opaque && view->container->alpha >= 1.0f && deco_data.alpha >= 1.0f)
+			&& config->blur_passes >= 0
+			&& config->blur_radius > 0) {
+		int width, height;
+		wlr_output_transformed_resolution(output->wlr_output, &width, &height);
+		struct wlr_fbox src_box = {
+			.x = 0,
+			.y = 0,
+			.width = width,
+			.height = height,
+		};
+
+		// Use the pre-rendered blurred buffer for tiled windows due tiled
+		// that all tiled windows are displaying the same content under the windows
+		bool is_floating = container_is_floating(view->container);
+		render_blur(!is_floating, output, output_damage, &src_box, &dst_box,
+				&surface->opaque_region, surface->current.width, surface->current.height,
+				surface->current.scale, deco_data, config->blur_radius, config->blur_passes);
```

Comment:
we could have a blur_config struct (similar to the border struct) that holds blur radius and blur passes to clean up our arguments